library/Asana/Api/Task.hs

```diff
@@ -150,28 +153,53 @@ instance FromJSON Task where
 getTask :: Gid -> AppM ext Task
 getTask taskId = getSingle $ "/tasks/" <> T.unpack (gidToText taskId)
 
+data TaskStatusFilter = IncompletedTasks | AllTasks
+  deriving (Show, Eq)
+
+newtype OptField = OptField
+  { getOptField :: Text
+  }
+  deriving (Show, Eq)
+
+data TaskSchema a = TaskSchema
+  { taskStatusFilter :: TaskStatusFilter
+  , taskOptFields :: [OptField]
+  }
+  deriving (Show, Eq)
+
+instance Semigroup (TaskSchema a) where
+  tsa <> tsb = tsa { taskOptFields = taskOptFields tsa <> taskOptFields tsb }
+
+instance Monoid (TaskSchema a) where
+  mempty = TaskSchema { taskStatusFilter = AllTasks, taskOptFields = [] }
```

Comment:
If you have this:

```haskell
data TaskSchema a = TaskSchema
  { taskStatusFilter :: Last TaskStatusFilter
  , taskOptFields :: [OptField]
  }
  deriving (Show, Eq)
```

All fields can "participate" in the `Semigroup` and `Monoid` instances